src/dns/server.rs

```diff
@@ -756,8 +803,17 @@ impl SystemForwarder {
 
 #[async_trait::async_trait]
 impl Forwarder for SystemForwarder {
-    fn search_domains(&self, _: &Workload) -> Vec<Name> {
-        self.search_domains.clone()
+    fn search_domains(&self, wl: &Workload) -> Vec<Name> {
+        // TODO: https://github.com/istio/ztunnel/issues/555 really get this from pods
+        // Today we hardcode the default search assumptions
+        match &self.search_domains {
+            SearchDomains::Static(s) => s.clone(),
+            SearchDomains::Dynamic(cluster_domain) => vec![
+                Name::from_utf8(format!("{}.svc.{cluster_domain}", wl.namespace)).unwrap(),
+                Name::from_utf8(format!("svc.{cluster_domain}")).unwrap(),
+                Name::from_utf8(cluster_domain).unwrap(),
```

Comment:
unwrap seems like it's probably not particularly unsafe but we should probably use expect or explicitly handle sad paths.

Reply:
:+1: `.expect` preferred simply because it's easier to read - the justification for why it should never fail is right there.